src/embind/embind.js

```diff
@@ -608,53 +611,80 @@ var LibraryEmbind = {
     '$simpleReadValueFromPointer', '$throwBindingError'],
   _embind_register_std_string: function(rawType, name) {
     name = readLatin1String(name);
+    var embindStdStringUTF8Support
+#if EMBIND_STD_STRING_UTF8_SUPPORT
+    = true;
+#else
+    = false;
+#endif
     registerType(rawType, {
         name: name,
         'fromWireType': function(value) {
             var length = HEAPU32[value >> 2];
-            var a = new Array(length);
-            for (var i = 0; i < length; ++i) {
-                a[i] = String.fromCharCode(HEAPU8[value + 4 + i]);
+
+            function mapsToStdString(name) {
```

Comment:
I would probably just add a new parameter to _embind_register_std_string, personally, but it's no big deal.

Reply:
Please see change around line 630 (introduction of the 'enableStdStringUTF8Support' variable), The decision has been moved to _embind_register_std_string(). Moving the decision further outside _embind_register_std_string(), using an additional parameter, the current call of _embind_register_std_string in bind.cpp line 115 would get a bit cluttered, don't you think?